### PR TITLE
[exporter/elasticsearch] Implement elasticsearch.mapping.hints attribute handling for data points in OTel mapping mode

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

Supersedes #35348 

elasticsearch.mapping.hints takes a slice of strings.
`_doc_count` enables emitting `_doc_count` for the document.
`aggregate_metric_double` causes histogram or exponential histogram to be emitted as aggregate_metric_double.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/elasticsearchexporter/mapping_hint.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+type mappingHint string
+
+const (
+	hintAggregateMetricDouble mappingHint = "aggregate_metric_double"
+	hintDocCount              mappingHint = "_doc_count"
+)
+
+func hasHint(attrs pcommon.Map, hint mappingHint) (found bool) {
```
Comment (lahsivjar) [inline exporter/elasticsearchexporter/mapping_hint.go:16]: This pattern is forcing us to iterate over the attribute every time we want to process a hint. I would prefer either processing all the hints together OR if that is not possible then extracting the hints in a separate struct and using it for checks whenever required.

Comment (carsonip) [reply exporter/elasticsearchexporter/mapping_hint.go:16]: good point. I'll see how much extra effort is needed to preprocess to extract hints.

Comment (carsonip) [reply exporter/elasticsearchexporter/mapping_hint.go]: done. I haven't gone so far to use a bitset, but i believe slices.Contains should be efficient enough.

File: .chloggen/elasticsearchexporter_mapping-hint.yaml [hunk 2]
```diff
@@ -7,15 +7,18 @@ change_type: enhancement
 component: elasticsearchexporter
 
 # A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
-note: Emit _doc_count for metric documents in OTel mode when data point attribute _doc_count is true
```
Comment (carsonip) [inline .chloggen/elasticsearchexporter_mapping-hint.yaml:10]: [to reviewer] Removing changelog for previous doc count PR as this PR supersedes it. It does not make sense for the doc count PR changelog to be in the release notes of the next version if it is not relevant.

**Review by mwear — APPROVED** [review]

**Review by lahsivjar — APPROVED** [review]
LGTM!

**Review by felixbarny — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]
I'm a bit worried that there's new functionality being added with no mention of it in the component's documentation. Isn't this functionality something users would be interested in?

**Pull request merged**